### ISSUE TITLE
[v3.0.0 build on Fedora 40] undefined symbol: xdp_input_capture_session_release

### ISSUE THREAD

**Issue by rmacd** [input-leap/input-leap#1991]
### What happened?

Installed `input-leap`, attempted to run and got below error. Installed `debuginfo` and `debugsource` and error remained the same.

log:
```
[2024-10-05T06:47:03] INFO: starting client
[2024-10-05T06:47:03] INFO: config file: /tmp/InputLeap.lKiLmy
[2024-10-05T06:47:03] INFO: log level: INFO
/usr/bin/input-leapc: symbol lookup error: /usr/bin/input-leapc: undefined symbol: xdp_input_capture_session_release
[2024-10-05T06:47:03] ERROR: process exited with error code: 127
[2024-10-05T06:47:03] INFO: detected process not running, auto restarting
```

`uname -a`:
```
Linux rm-p1 6.10.10-200.fc40.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Sep 12 18:26:09 UTC 2024 x86_64 GNU/Linux
```

`/etc/fedora-release`: Fedora release 40 (Forty)

### Version

v3.0.0

### Git commit hash (if applicable)

build: ab1c0f5 run: [`11172482132`](https://github.com/input-leap/input-leap/actions/runs/11172482132) artifact: [`input-leap-rpms-fedora`](https://github.com/input-leap/input-leap/actions/runs/11172482132/artifacts/2013831545)

### If applicable, where did you install Input Leap from?

GitHub action

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

Tested with Gnome on both Wayland and X11

### What OS versions are you using?

Fedora 40

### Relevant log output

```
[2024-10-05T06:47:03] INFO: starting client
[2024-10-05T06:47:03] INFO: config file: /tmp/InputLeap.lKiLmy
[2024-10-05T06:47:03] INFO: log level: INFO
/usr/bin/input-leapc: symbol lookup error: /usr/bin/input-leapc: undefined symbol: xdp_input_capture_session_release
[2024-10-05T06:47:03] ERROR: process exited with error code: 127
[2024-10-05T06:47:03] INFO: detected process not running, auto restarting
```

### Any other information

Error can be resolved by installing `libportal-devel` (flatpak). This suggests the package either erroneously depends on `libportal-devel`, or this dependency needs to be added to the specfile.

**Comment by Cimbali** [comment]
Can you check your version of  `libportal`? I think it needs to be 0.8.1 (but the rpm specifies >= 0.8 only).

```
> rpm -q libportal1
libportal1-0.8.1-1.1.x86_64
> nm -D /usr/lib64/libportal.so.1.0.0|grep xdp_input_capture_session_release
000000000000ee80 T xdp_input_capture_session_release
000000000000edf0 T xdp_input_capture_session_release_at
```

**Comment by KoljaFrahm** [comment]
There was no libportal 0.8 build other than 0.8.1 on Fedora, and I think it should also work with 0.8
Maybe we have to add a `Requires: libportal` to the spec file? @whot you said we don't need any Requires?

**Comment by Cimbali** [comment]
The symbol used was added in 0.8.1 which is why the CI built from master for a while, so the package does require 0.8.1. Fedora just isn’t up to date yet but that’s another problem.

**Comment by Cimbali** [comment]
https://github.com/input-leap/input-leap/blob/ab1c0f52996b2bad2e24f1eee69a46240451428e/dist/rpm/input-leap.spec.in#L34

**Comment by Cimbali** [comment]
You indeed need `libportal` and `libei` installed, not just for building. The binary is linked to those.

```
> ldd `which input-leapc` | grep -e libei -e libportal
    libei.so.1 => /lib64/libei.so.1 (0x00007f788bd6f000)
    libportal.so.1 => /lib64/libportal.so.1 (0x00007f788bb48000)
```

**Comment by rmacd** [comment]
> Can you check your version of  `libportal`? I think it needs to be 0.8.1 (but the rpm specifies >= 0.8 only).
> 
> ```
> > rpm -q libportal1
> libportal1-0.8.1-1.1.x86_64
> > nm -D /usr/lib64/libportal.so.1.0.0|grep xdp_input_capture_session_release
> 000000000000ee80 T xdp_input_capture_session_release
> 000000000000edf0 T xdp_input_capture_session_release_at
> ```

Installation candidate by default was `0.8.1`.

```
# yum info libportal
Last metadata expiration check: 0:05:21 ago on Sat 05 Oct 2024 12:04:23 BST.
Installed Packages
Name         : libportal
Version      : 0.8.1
Release      : 1.fc40
Architecture : x86_64
Size         : 216 k
Source       : libportal-0.8.1-1.fc40.src.rpm
Repository   : @System
From repo    : updates
Summary      : Flatpak portal library
URL          : https://github.com/flatpak/libportal
License      : LGPL-3.0-only AND LGPL-2.1-or-later
Description  : libportal provides GIO-style asynchronous APIs for most Flatpak portals.
```

**Comment by KoljaFrahm** [comment]
Just tried this with my Fedora 40 installation. libportal is already required by the dnf and I don't have to install libportal-devel for input-leap to work. Tested for client and server, and under wayland and x11.

**Comment by rmacd** [comment]
Potentially. This was not installed on a clean system.

**Comment by KoljaFrahm** [comment]
Then this issue can potentially be closed (and opened again if it also happens with libportal 0.8 for someone but I don't think so)

**Comment by p12tic** [comment]
Yeah, I think we can close the issue and reopen if it reappears.

@rmacd Thanks for the bug report. Even if this turns out not to be bug in Input Leap, other people with similar problem can find this page and possibly fix the problem for themselves.

**Comment by Gorlinux** [comment]
Got the same issue on Nobara (Fedora FC40 + KDE + Wayland).

Saw that most available libportal versions are 0.7.1-4.fc40 execept one:
```
root@x9b:/home/gregor# dnf install  libportal* 2>/dev/null | grep libportal
 libportal-debuginfo      x86_64 0.7.1-4.fc40   nobara-baseos-40          174 k
 libportal-debugsource    x86_64 0.7.1-4.fc40   nobara-baseos-40           60 k
 libportal-devel-doc      noarch 0.7.1-4.fc40   nobara-baseos-40           96 k
 libportal-gtk3           i686   0.7.1-4.fc40   nobara-baseos-multilib-40  17 k
 libportal-gtk3-debuginfo x86_64 0.7.1-4.fc40   nobara-baseos-40           23 k
 libportal-gtk3-devel     i686   0.7.1-4.fc40   nobara-baseos-multilib-40  11 k
 libportal-gtk4           i686   0.7.1-4.fc40   nobara-baseos-multilib-40  17 k
 libportal-gtk4-debuginfo x86_64 0.7.1-4.fc40   nobara-baseos-40           23 k
 libportal-gtk4-devel     i686   0.7.1-4.fc40   nobara-baseos-multilib-40  11 k
 libportal-qt5            i686   0.7.1-4.fc40   nobara-baseos-multilib-40  27 k
 libportal-qt5-debuginfo  x86_64 0.7.1-4.fc40   nobara-baseos-40          201 k
 libportal-qt5-devel      i686   0.7.1-4.fc40   nobara-baseos-multilib-40  11 k
 libportal-qt6-devel      x86_64 0.8.1-1.fc40   updates                    10 k
 libportal                i686   0.7.1-4.fc40   nobara-baseos-multilib-40  77 k
 libportal-devel          i686   0.7.1-4.fc40   nobara-baseos-multilib-40  43 k
 libportal                x86_64 0.7.1-4.fc40   nobara-baseos-40           75 k
 libportal-devel          x86_64 0.7.1-4.fc40   nobara-baseos-40           43 k
 libportal-gtk3           x86_64 0.7.1-4.fc40   nobara-baseos-40           18 k
 libportal-gtk3-devel     x86_64 0.7.1-4.fc40   nobara-baseos-40           11 k
 libportal-gtk4           x86_64 0.7.1-4.fc40   nobara-baseos-40           18 k
 libportal-gtk4-devel     x86_64 0.7.1-4.fc40   nobara-baseos-40           11 k
 libportal-qt5            x86_64 0.7.1-4.fc40   nobara-baseos-40           27 k

```

Solution: 
sudo dnf install libportal-qt6
